benchmarks/__main__.py

```diff
@@ -81,17 +86,20 @@ def run_benchmarks(emails, results_dir, incremental):
         try:
             with Timer.timeit() as read_timer:
                 with compressor.open_read(outpath) as fobj:
-                    for _ in serializer.deserialize(fobj):
-                        pass
+                    with encryptor.deserialize(fobj) as dec:
```

Comment:
From my understanding, modern encryption transforms the incoming bytes into an essentially random byte string (e.g. to prevent statistical frequency analysis attacks). As such, I believe the encryption should happen after the compression since otherwise the compression won't have enough signal/patterns in the encrypted byte string to produce good results. Am I mistaken here?

Reply:
Hmmm I actually have no Idea but that makes sense! I'll look more into this!

Reply:
Just looked into this and yes, compression before encryption is the recommended course of action.